<!DOCTYPE html>
<html>
<head>
  <!-- Standard Meta -->
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

  <title>Benchmarking LIRICAL for Phenotype-based Prioritization In 75 Rare Disease Cases</title>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css" integrity="sha512-8bHTC73gkZ7rZ7vpqUQThUDhqcNFyYi2xgDgPDHc+GXVGHXq+xPjynxIopALmOPqzo9JZj0k6OqqewdGO3EsrQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-G7WEXHL9KW"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-G7WEXHL9KW');
  </script>
  <style>
    .header-section {
      font-weight: 500;
      color: #555555;
      display: inline-block;
    }

    /* used by anchor-js */
    :target {
      scroll-margin-top: 70px;
    }

  </style>
</head>
<body>

<a name="body-top"></a>

<div class="ui fixed menu" style="padding: 10px 0px">
  <div class="ui grid" style="width:100%; font-size:16px; color: rgba(0,0,0,.87); font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif">
    <div class="ui three wide column"></div>
    <div class="ui three wide column">
      <div style="font-weight:800; padding-right:10px; display:inline-block"><a href="/">Project Notes</a></div>
      <div class="header-section"> - Ben Weisburd</div>
    </div>
    <div class="ui six wide column">
      <div class="header-section, header-section-fade-in" style="color: black; display: none; width:100%">Benchmarking LIRICAL for Phenotype-based Prioritization In 75 Rare Disease Cases</div>
    </div>
    <div class="ui one wide column">
      <div class="header-section" style="float: right;">5/4/2022</div>
    </div>
    <div class="ui three wide column">
      <div class="header-section, header-section-fade-in" style="display: none;">[<a href="#body-top">Jump to top</a>]</div>
    </div>
  </div>
</div>

<div class="ui grid">
  <div class="ui three wide column"></div>
  <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif; padding-top:35px">

    <h2>Benchmarking LIRICAL for Phenotype-based Prioritization In 75 Rare Disease Cases</h2>
<p>LIRICAL [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7477017/">Robinson 2020</a>] is a phenotype-based gene &amp; variant prioritization tool.</p>
<p>Let's say an individual has symptoms described by these 7 human phenotype ontology (HPO) terms:</p>
<ul>
<li>Hearing impairment (<a href="https://hpo.jax.org/app/browse/term/HP:0000365">HP:0000365</a>)  </li>
<li>Autism (<a href="https://hpo.jax.org/app/browse/term/HP:0000717">HP:0000717</a>)</li>
<li>Seizure (<a href="https://hpo.jax.org/app/browse/term/HP:0001250">HP:0001250</a>)</li>
<li>Global developmental delay (<a href="https://hpo.jax.org/app/browse/term/HP:0001263">HP:0001263</a>)</li>
<li>Gastrointestinal dysmotility (<a href="https://hpo.jax.org/app/browse/term/HP:0002579">HP:0002579</a>)</li>
<li>Short stature (<a href="https://hpo.jax.org/app/browse/term/HP:0004322">HP:0004322</a>)</li>
<li>Supraventricular tachycardia (<a href="https://hpo.jax.org/app/browse/term/HP:0004755">HP:0004755</a>)</li>
</ul>
<p>The LIRICAL algorithm takes these as input along with the individual's variant call file (from WGS or WES) and outputs a list of candidate diagnoses, 
their post-test probabilities, and visualizations showing which of the individual's HPO terms support or contradict each candidate.</p>
<p>For the above example, LIRICAL outputs:</p>
<p><img width="1000" alt="LIRICAL output screenshot" src="https://user-images.githubusercontent.com/6240170/166859374-128a44d3-6590-44e4-ba95-279bfa2fada9.png"></p>
<p>Clinical geneticists that previously evaluated this case without using LIRICAL came to the conclusion that variants in the KMT2C gene explain 
this patient's phenotype. </p>
<p>To evaluate LIRICAL performance, I ran it on 75 previously-solved rare disease cases from the <a href="https://raregenomes.org/">Rare Genomes Project</a>. </p>
<ul>
<li>59 cases: solved with a known gene-disease association. </li>
<li>2 cases: solved with a gene that was previously associated with a different disease phenotype. </li>
<li>14 cases: solved with a gene not previously associated with a disease. </li>
</ul>
<p>LIRICAL identified the correct gene in 65 out of the 75 cases (87%), and in 42 out of 75 cases (56%) the correct gene was in the top 3 results.</p>
<p><i><u><b>NOTE</b></u>:  By default, LIRICAL uses gnomAD v2 allele frequencies (AFs) when analyzing variants. In order to take advantage of gnomAD v3.1 AFs, I prefilted all variant call files (VCFs) to gnomAD v3.1 POPMAX AF &lt; 1% before passing them to LIRICAL. See the last section below for stats on how this improved results.</i></p>
<p><br />
This histogram shows how often LIRICAL ranked the correct gene in the top 5, between 6-10, and &gt;10th in the list, colored by inheritance mode. 
The x-axis represents # of cases: <br /> </p>
<p><br />
<img width="1000" alt="LIRICAL correct gene rank by inheritance mode" src="https://user-images.githubusercontent.com/6240170/169574825-3e719744-07fe-4e0e-afc6-29c856436c27.png"></p>
<p>Another way to look at this is - for the 65 cases where LIRICAL found the correct gene, what was that gene's rank in the 
list and it's post-test probability:</p>
<p><img width="750" alt="Plot post-test probability vs LIRICAL rank" src="https://user-images.githubusercontent.com/6240170/166860896-d35babad-a1a5-4d20-a9b4-cc3f58dc8045.png"></p>
<p>To summarize this plot, in 49 out of 75 cases (65%) the correct result was among the top 5 hits and for 36 out of 75 (48%) it also had a post-test probability between 1% and 100%. On the other hand, for 16 cases (21%) the correct result was below 5th in the list and had a post-test probability near 0. For another 10 cases (13%), the correct result was not in the list at all. </p>
<p><strong>Conclusion:</strong> LIRICAL shows relatively high sensitivity in finding the causal genes for rare disease cases from <a href="https://raregenomes.org/">RGP</a>. </p>
<hr />
<p><strong>How many genes would someone need to look through for a typical case?</strong></p>
<p>From the above plot, a conservative approach would be to only consider the top 5 results (x-axis cutoff) and only if they have a post-test probability (PTP) of 1% or higher (y-axis cutoff). This would give 3 results (or more percisely, 2.53 results) to look at on average, with 24% of cases having 5 or more results with PTP &gt; 1%:</p>
<p><img width="500" alt="histogram: % of RGP cases that have 0, 1, 2, 3, 4, or 5 LIRICAL results with PTP > 1 and rank &lt;= 5" src="https://user-images.githubusercontent.com/6240170/169594097-7f9caeee-824c-4baf-b5d4-0992dfc70c56.png"&gt;</p>
<hr />
<p><strong>What is the false-positive rate?</strong></p>
<p>The above plot shows that, although LIRICAL reports the correct gene as #1 in the list in 23 out of 75 cases (31%), most of the time the 
correct gene is further down in the list. All genes except the correct gene can be considered false positives. With conservative thresholds of only looking at the top 5 results and only when they have a post-test probability (PTP) of 1% or higher, analysts would need to look at ~2 false-positive genes per case on average.</p>
<p>To test LIRICAL performance another way and also see how much it depends on variant-based matching vs. phenotype-based matching, I reran the same cases, but substituted in variants from completely unrelated healthy individuals sequenced as part of the same cohort. I did this 5x for each of the 75 solved cases above (=&gt; 375 test runs). With VCFs from healthy individuals, LIRICAL accuracy dropped to esssentially 0: only 10 out of the 375 LIRICAL reports (3%) still included the correct gene, and only 3 of these were in the top 5 and had a post-test probability &gt; 1%.</p>
<p>However, in 260 out of 375 tests (70%) LIRICAL still reported at least one result with a post-test probability &gt; 1%. This again shows that, even with conservative thresholds of top-5 and PTP &gt; 1%, a user should expect to see on average 2 false-positive results per case:</p>
<!-- img width="429" alt="image" src="https://user-images.githubusercontent.com/6240170/166864705-af5f4943-d41e-4416-967d-e6f16183c85a.png" -->
<p><img width="500" alt="histogram: % of RGP cases that had 0, 1, 2, 3, 4, or 5 false-positive LIRICAL results with PTP > 1 and rank &lt;= 5" src="https://user-images.githubusercontent.com/6240170/166866744-55f075b5-08e5-4999-bbec-8bde066add03.png"&gt;</p>
<p>For comparison, in the first analysis, 62 out of 75 cases (83%) had at least one result with a post-test probability &gt; 1%, so LIRICAL post-test probability is on average slightly lower for false-positive results. </p>
<hr />
<p><strong>Can we do better by dropping the post-test probability threshold to 0%?</strong></p>
<p>Previously we saw that for 36 out of 75 (48%) cases, the correct gene was ranked in the top 5 and had a post-test probability between 1% and 100%. 
Would we get higher sensitivity/specificity if we used different thresholds? The answer is yes:</p>
<table class="ui striped center aligned table">
  <tr><th>Post-test probability threshold</td><th>Top-k results</td><th># of true-positive genes <br />passing the thresholds</th><th># of genes (per case)<br />passing the thresholds</td></tr>
  <tr><td>&gt; 1%</td><td>top 5 genes</td><td>36 out of 75 (48%)</td><td>190/75 = 2.5 genes</td></tr>
  <tr><td>&gt; 0%</td><td>top 5 genes</td><td>49 out of 75 (65%)</td><td>287/75 = 3.8 genes</td></tr>
  <tr><td>&gt; 0%</td><td>top 4 genes</td><td>45 out of 75 (60%)</td><td>239/75 = 3.2 genes</td></tr>
  <tr><td>&gt; 0%</td><td>top 3 genes</td><td>42 out of 75 (56%)</td><td>186/75 = 2.5 genes</td></tr>
  <tr><td>&gt; 0%</td><td>top 2 genes</td><td>40 out of 75 (53%)</td><td>129/75 = 1.7 genes</td></tr>
  <tr><td>&gt; 0%</td><td>top 1 genes</td><td>29 out of 75 (39%)</td><td>&nbsp;68/75 = 0.9 genes</td></tr>
</table>

<p>Based on this, it's better to look at the top 3 genes with PTP &gt; 0% instead of the top 5 and PTP &gt; 1%.</p>
<hr />
<p><strong>Can LIRICAL performance be improved further?</strong></p>
<p>Yes - through additional prefiltering. For the tests above, I prefiltered each individual's variants to exclude variants that are common in the general population (keeping only variants with gnomAD v3 PopMax AF &lt; 0.01). Applying this prefilter didn't change the number of true positive results, but improved their average rank from 7.6 to 5.9. Adding further prefiltering - such as by inheritance mode for cases where parents' DNA is available - will almost certainly improve results further. </p>
<hr />
<p><strong>References:</strong></p>
<ol>
<li>Robinson PN, Ravanmehr V, Jacobsen JOB, et al. Interpretable Clinical Genomics with a Likelihood Ratio Paradigm. Am J Hum Genet. 2020;107(3):403-417. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7477017/">doi:10.1016/j.ajhg.2020.06.021</a></li>
<li>Yuan X, Wang J, Dai B, et al. Evaluation of phenotype-driven gene prioritization methods for Mendelian diseases. Brief Bioinform. 2022;23(2):bbac019. <a href="https://pubmed.ncbi.nlm.nih.gov/35134823/">doi:10.1093/bib/bbac019</a></li>
</ol>
<p><strong>Source code:</strong></p>
<ul>
<li>LIRICAL pipeline for <a href="https://hail.is/docs/batch/service.html">Hail Batch</a>: <br />
<a href="https://github.com/broadinstitute/phenotype-driven-analysis">https://github.com/broadinstitute/phenotype-driven-analysis</a></li>
</ul>

    <br />
    <center><img style="margin-left:25px;" width="586" alt="image" src="https://user-images.githubusercontent.com/6240170/167290392-8974f18a-af18-40c1-9f61-fbb6713fc8d0.png"></center>

    <!-- https://giscus.app/ -->
    <div class="container giscus" style="width:100%; margin:50px 10px">
      <script src="https://giscus.app/client.js"
              data-repo="bw2/bw2.github.io"
              data-repo-id="MDEwOlJlcG9zaXRvcnkzNTU2OTQ0OA=="
              data-category="Announcements"
              data-category-id="DIC_kwDOAh6_KM4CO_Vz"
              data-mapping="pathname"
              data-reactions-enabled="1"
              data-emit-metadata="0"
              data-input-position="bottom"
              data-theme="light"
              data-lang="en"
              crossorigin="anonymous"
              async>
      </script>
    </div>
  </div>
  <div class="ui three wide column"></div>
</div>

<script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"  integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js" integrity="sha512-dqw6X88iGgZlTsONxZK9ePmJEFrmHwpuMrsUChjAw1mRUhUITE5QU9pkcSox+ynfLhL15Sv2al5A0LVyDCmtUw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script>
  const headingAnchors = new AnchorJS({
    visible: 'hover',
    placement: 'left'
  });
  headingAnchors.add('h2, h3, h4');

  const paragraphAnchors = new AnchorJS({
    visible: 'hover',
    placement: 'left',
    icon: 'Â¶'
  });
  paragraphAnchors.add('p');

  $( window ).scroll(() => {
    if (window.scrollY > 75) {
      $(".header-section-fade-in").fadeIn("slow");
    } else {
      $(".header-section-fade-in").fadeOut("slow");
    }
  })
</script>
</body>
</html>