STR calling tools like ExpansionHunter and GangSTR require users to specify the motifs and coordinates of all loci to genotype. 
For genome-wide STR calling, the choice of loci sets an upper-bound on the tool's sensitivity since the tools will check for STRs only at these loci. 

To bechmark GangSTR and ExpansionHunter and allow for maximum sensitivity, we created the input set of loci by following these steps:

1. 
