<!DOCTYPE html>
<html>
<head>
  <!-- Standard Meta -->
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

  <title>How adjacent repeats in ExpansionHunter catalogs affect genotype quality</title>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css" integrity="sha512-8bHTC73gkZ7rZ7vpqUQThUDhqcNFyYi2xgDgPDHc+GXVGHXq+xPjynxIopALmOPqzo9JZj0k6OqqewdGO3EsrQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
  <link rel="stylesheet" href="./github-css.css" />
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-G7WEXHL9KW"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-G7WEXHL9KW');
  </script>
  <style>
    .header-section {
      font-weight: 500;
      color: #555555;
      display: inline-block;
    }

    /* used by anchor-js */
    :target {
      scroll-margin-top: 70px;
    }

    pre code {
      background-color: #F8F8F8 !important;
      padding: 12px !important;
      border: none !important;
      display: block;
    }

  </style>
</head>
<body>

<a name="top"></a>

<div class="ui fixed menu" style="padding: 10px 0px">
  <div class="ui grid" style="width:100%; font-size:16px; color: rgba(0,0,0,.87)">
    <div class="ui three wide column"></div>
    <div class="ui three wide column">
      <div style="font-weight:800; padding-right:10px; display:inline-block"><a href="/">Project Notes</a></div>
      <div class="header-section"> - Ben Weisburd</div>
    </div>
    <div class="ui six wide column">
      <div class="header-section, header-section-fade-in" style="color: black; display: none; width:100%">How adjacent repeats in ExpansionHunter catalogs affect genotype quality</div>
    </div>
    <div class="ui one wide column">
      <div class="header-section" style="float: right;">7/31/2023</div>
    </div>
    <div class="ui three wide column">
      <div class="header-section, header-section-fade-in" style="display: none;">[<a href="#top">Jump to top</a>]</div>
    </div>
  </div>
</div>

<div class="ui grid">
  <div class="ui three wide column"></div>
  <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px;">

    <h2>How adjacent repeats in ExpansionHunter catalogs affect genotype quality</h2>
<p>ExpansionHunter has a unique feature where, for each locus, users can optionally specify adjacent STR loci, and ExpansionHunter will genotype both the main locus and the adjacent loci together. For example, for the Huntington's disease locus, rather than just specifying the CAG repeat as in:</p>
<pre><code>{
    "LocusId": "HTT",
    "LocusStructure": "(CAG)*",
    "ReferenceRegion": "4:3074876-3074933",
    "VariantType": "Repeat",
}</code></pre>
<p>the user can also specify the adjacent CCG repeat (source: 
<a href="https://github.com/Illumina/ExpansionHunter/blob/master/variant_catalog/grch38/variant_catalog.json#L458-L473">the official ExpansionHunter catalog</a>):</p>
<pre><code>{
    "LocusId": "HTT",
    "LocusStructure": "(CAG)*CAACAG(CCG)*",
    "ReferenceRegion": [
        "4:3074876-3074933",
        "4:3074939-3074966"
    ],
    "VariantId": [
        "HTT",
        "HTT_CCG"
    ],
    "VariantType": [
        "Repeat",
        "Repeat"
    ]
}</code></pre>

<p>To understand how specifying adjacent loci affects ExpansionHunter performance, I wrote a script that adds adjacent 
loci to existing STR catalogs. It takes any ExpansionHunter variant catalog (such as the 
Illumina <a href="https://github.com/Illumina/RepeatCatalogs">catalog of 174k polymorphic loci</a>) as well as a catalog of all 
STRs in the reference genome (such as those produced by running TandemRepeatFinder) and outputs a new catalog after 
adding adjacent repeats that are within a user-defined distance from the original locus. This script is 
called <code>add_adjacent_loci_to_expansion_hunter_catalog</code> and is available in the 
<a href="https://github.com/broadinstitute/str-analysis">str_analysis</a> repo.</p>
<h3>TR truth set analysis</h3>
<p>To test this script and evaluate how adjacent loci affect ExpansionHunter accuracy, I used the TR truth set from 
<a href="https://www.biorxiv.org/content/10.1101/2023.05.05.539588v1">Insights from a genome-wide truth set of tandem repeat variation by Ben Weisburd, Grace Tiao, Heidi L. Rehm
bioRxiv 2023.05.05.539588</a>. </p>
<p>This truth set contains 146,318 TR loci that are polymorphic in the CHM1-CHM13 synthetic diploid sample and also present in the hg38 reference.</p>
<p>I converted these to ExpansionHunter catalog format, and then  ran <code>add_adjacent_loci_to_expansion_hunter_catalog</code> on it. For the reference TR catalog argument, I used </p>
<pre><code>gs://str-truth-set/hg38/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_9bp.bed.gz</code></pre>
<p>which contains 4,484,369 pure TR loci detected in hg38 by TandemRepeatFinder.<br />
Also, I set the <code>--max-distance-between-adjacent-repeats</code> parameter to 50bp which is 1/3 the read length. </p>
<p>The full command line was:</p>
<pre><code>python3 -u -m str_analysis.add_adjacent_loci_to_expansion_hunter_catalog \
    --source-of-adjacent-loci ./ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_9bp.bed.gz \
    --ref-fasta ./ref/hg38.fa \
        --max-distance-between-adjacent-repeats 50 \
    --add-extra-fields \
    truth_set_variant_catalog.json</code></pre>

<h3>Adjacent loci summary statistics</h3>
<p>For the 146,318 polymorphic TR loci in CHM1-CHM13:</p>
<pre><code>  67,003 out of 146,318 polymorphic loci (45.8%) were found to have 1 or more adjacent repeats</code></pre>

<p>The distances between the adjacent loci were distributed as follows:</p>
<pre><code>  39,850 out of  114,220 spacers ( 34.9%) had     0bp spacer
   8,772 out of  114,220 spacers (  7.7%) had     1bp spacer
   3,895 out of  114,220 spacers (  3.4%) had     2bp spacer
   4,240 out of  114,220 spacers (  3.7%) had     3bp spacer
   3,148 out of  114,220 spacers (  2.8%) had     4bp spacer
   3,481 out of  114,220 spacers (  3.0%) had     5bp spacer
   ...
   2,189 out of  114,220 spacers (  1.9%) had    10bp spacer
   ...
   1,952 out of  114,220 spacers (  1.7%) had    15bp spacer
   ...
   1,202 out of  114,220 spacers (  1.1%) had    20bp spacer
   ...
     687 out of  114,220 spacers (  0.6%) had    30bp spacer
   ...
     516 out of  114,220 spacers (  0.5%) had    40bp spacer
   ...
     393 out of  114,220 spacers (  0.3%) had    50bp spacer</code></pre>

<p>The most common configuration was a pair of adjacent loci (the main locus + a newly added adjacent locus) with differing motifs - like the HTT locus example above. The other configurations were as follows:</p>
<pre><code>  30,029 out of  260,538 reference regions ( 11.5%) had adjacent repeats count:   2 reference regions with different motifs
   9,907 out of  260,538 reference regions (  3.8%) had adjacent repeats count:   2 reference regions with same motif
  15,117 out of  260,538 reference regions (  5.8%) had adjacent repeats count:   3 reference regions with different motifs
     631 out of  260,538 reference regions (  0.2%) had adjacent repeats count:   3 reference regions with same motif
   6,286 out of  260,538 reference regions (  2.4%) had adjacent repeats count:   4 reference regions with different motifs
      74 out of  260,538 reference regions (  0.0%) had adjacent repeats count:   4 reference regions with same motif
   2,709 out of  260,538 reference regions (  1.0%) had adjacent repeats count:   5 reference regions with different motifs
       4 out of  260,538 reference regions (  0.0%) had adjacent repeats count:   5 reference regions with same motif
   1,281 out of  260,538 reference regions (  0.5%) had adjacent repeats count:   6 reference regions with different motifs
     556 out of  260,538 reference regions (  0.2%) had adjacent repeats count:   7 reference regions with different motifs
     247 out of  260,538 reference regions (  0.1%) had adjacent repeats count:   8 reference regions with different motifs
     100 out of  260,538 reference regions (  0.0%) had adjacent repeats count:   9 reference regions with different motifs
      43 out of  260,538 reference regions (  0.0%) had adjacent repeats count:  10 reference regions with different motifs
      14 out of  260,538 reference regions (  0.0%) had adjacent repeats count:  11 reference regions with different motifs
       2 out of  260,538 reference regions (  0.0%) had adjacent repeats count:  12 reference regions with different motifs
       2 out of  260,538 reference regions (  0.0%) had adjacent repeats count:  13 reference regions with different motifs
       1 out of  260,538 reference regions (  0.0%) had adjacent repeats count:  14 reference regions with different motifs</code></pre>

<p>Unsurprisingly, if I went back and selected approximately the same number of non-polymorphic loci (145,429) from the hg38 reference so that they had the same distribution of motif sizes as the polymorphic loci, a smaller percentage of the non-polymorphic lcoi had adjacent repeats:</p>
<pre><code>   49,862 out of 145,429 non-polymorphic loci (34.4%) were found to have 1 or more adjacent repeats</code></pre>

<h3>ExpansionHunter results with vs. without specifying adjacent loci</h3>
<p>To see how ExpansionHunter performance changes when specifying adjacent loci, I followed these steps:
- selected the 66,523 (45.5%) of polymorphic loci that had ExpansionHunter calls and at least 1 adjacent repeat within 50bp of the main locus 
- generated 2 ExpansionHunter variant catalogs for these 66,523 loci: one that included adjacent repeats, and one that didn't
- looked at differences between ExpansionHunter calls with vs. without adjacent repeats</p>
<pre><code> 57,854 out of 66,523 (87%) loci had the same exact genotype with vs. without specifying adjacent loci</code></pre>

<p>To see how different distances (ie. spacer sizes) between adjacent repeats affected results, I selected subsets of 
loci where the adjacent repeats were closer together than a particular max distance, and checked
what fraction of them had the exact same ExpansionHunter genotype with vs. without specifying adjacent repeat(s):</p>
<pre><code>Distance between adjacent repeats =   0bp: Genotype didn't change for 11,918 out of 15,083 (79.0%) loci
Distance between adjacent repeats <= 10bp: Genotype didn't change for 30,495 out of 36,224 (84.2%) loci
Distance between adjacent repeats <= 20bp: Genotype didn't change for 41,863 out of 48,928 (85.6%) loci
Distance between adjacent repeats <= 30bp: Genotype didn't change for 48,826 out of 56,669 (86.2%) loci
Distance between adjacent repeats <= 40bp: Genotype didn't change for 53,727 out of 62,035 (86.6%) loci
Distance between adjacent repeats <= 50bp: Genotype didn't change for 57,854 out of 66,523 (87.0%) loci</code></pre>

<p>When plotted, this looks like:  </p>
<p><img width="577" alt="image" src="https://github.com/bw2/bw2.github.io/assets/6240170/449ab132-da02-42aa-a574-52a9c3822345">
<br />
<br />
This suggests that there's not much point in setting --max-distance-between-adjacent-repeats to values larger than ~20bp.</p>
<h3>Concordance with truth set genotypes</h3>
<p>I then checked ExpansionHunter accuracy as measured by concordance with the true genotypes from the Synthetic Diploid Benchmark:</p>
<p><img width="851" alt="image" src="https://github.com/bw2/bw2.github.io/assets/6240170/fa58e8d6-e64e-4928-83d7-6232dfdff138"></p>
<p>The blue line represents the calls with adjacent loci, and is slightly lower - suggesting that specifying adjacent loci <strong>slightly reduced</strong> concordance with the truth set. </p>
<p>More detailed stats are below, where genotyping error at a locus is defined as the total difference between the true allele size and ExpansionHunter's estimated allele size for allele1 + allele2.</p>
<pre><code>57,854 out of 66,523 (  87%) loci: genotype error didn't change when running ExpansionHunter with vs. without adjacent loci.
---
 2,360 out of 66,523 ( 3.5%) loci: allele1 + allele2 genotype error decreased by at least 3 repeats when using adjacent loci.
 1,321 out of 66,523 ( 2.0%) loci: allele1 + allele2 genotype error INCREASED by at least 3 repeats when using adjacent loci.
---
 1,843 out of 66,523 ( 2.8%) loci: allele1 + allele2 genotype error decreased by at least 5 repeats when using adjacent loci.
   832 out of 66,523 ( 1.3%) loci: allele1 + allele2 genotype error INCREASED by at least 5 repeats when using adjacent loci.</code></pre>

<p>These proportions between the number of loci where errors increased vs. decreased stay approximately the same even when I prefilter to alleles that have the highest genotype 
quality scores (Q &gt; 0.8), or those that only have trinucleotide motifs. </p>
<h3>Manual review of read visualizations</h3>
<p>I then looked at REViewer images for 100 loci where adding adjacent repeats changed the genotype by 3 or more repeats total. 
For this, I selected only pure repeat loci where adjacent repeats didn't have any spacers between them (since on initial review, it looked like these were the loci where genotype quality 
improved most clearly after the addition of adjacent repeats). 
At each locus, I compared two images: one from running ExpansionHunter with adjacent repeats specified, and one without adjacent repeats. </p>
<p>The stats for these 100 loci were:</p>
<pre><code>50 loci had low quality genotypes both with and without repeats
36 loci had higher quality genotypes when adjacent repeats were specified
10 loci had lower quality genotypes when adjacent repeats were specified
 4 loci were amgibuous</code></pre>

<p>For 20 loci, the true genotype from the STR truth set appeared to be incorrect - implying that <strong>loci where ExpansionHunter genotypes significantly differ with vs. without adjacent repeats are enriched for assembly errors or cell line mutations in the CHM1-CHM13 synthetic diploid</strong>. </p>
<p>The 200 REViewer images for these 100 loci can be viewed here:  [<a href="https://bw2.github.io/2023-07-31_adjacent_repeats_flipbook/">200 REViewer images</a>]</p>
<h3>Conclusions</h3>
<ul>
<li>It's possible to find/add adjacent repeats within 50bp of the main locus for 30 to 50% of loci in any given STR catalog. </li>
<li>Within those 30 to 50%, adjacent repeats only make a difference (ie. change ExpansionHunter's genotype) in 10% to 20% of loci.</li>
<li>Overall, for 2.5% of polymorphic loci (2,360 + 1,321 out of 146,318), adding adjacent repeats changed the ExpansionHunter genotype by 3 or more repeats.</li>
<li>For these 2.5% of loci, introducing adjacent repeats improved the genotype quality for 36% of them, and reduced the genotype quality in 10% of them (based on manual review). For 50% of these loci, the genotype quality remained equally bad with and without adjacent repeats.</li>
<li>When running <code>add_adjacent_loci_to_expansion_hunter_catalog.py</code>, 20 is a reasonable value for <code>--max-distance-between-adjacent-repeats</code>.</li>
</ul>

    <br />
    <center><img style="margin-left:25px;" width="586" alt="image" src="https://user-images.githubusercontent.com/6240170/167290392-8974f18a-af18-40c1-9f61-fbb6713fc8d0.png"></center>

    <!-- https://giscus.app/ -->
    <div class="container giscus" style="width:100%; margin:50px 10px">
      <script src="https://giscus.app/client.js"
              data-repo="bw2/bw2.github.io"
              data-repo-id="MDEwOlJlcG9zaXRvcnkzNTU2OTQ0OA=="
              data-category="Announcements"
              data-category-id="DIC_kwDOAh6_KM4CO_Vz"
              data-mapping="pathname"
              data-reactions-enabled="1"
              data-emit-metadata="0"
              data-input-position="bottom"
              data-theme="light"
              data-lang="en"
              crossorigin="anonymous"
              async>
      </script>
    </div>
  </div>
  <div class="ui three wide column"></div>
</div>

<script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"  integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js" integrity="sha512-dqw6X88iGgZlTsONxZK9ePmJEFrmHwpuMrsUChjAw1mRUhUITE5QU9pkcSox+ynfLhL15Sv2al5A0LVyDCmtUw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script>
  const headingAnchors = new AnchorJS({
    visible: 'hover',
    placement: 'left'
  });
  headingAnchors.add('h2, h3, h4');

  const paragraphAnchors = new AnchorJS({
    visible: 'hover',
    placement: 'left',
    icon: 'Â¶'
  });
  paragraphAnchors.add('p');

  $( window ).scroll(() => {
    if (window.scrollY > 75) {
      $(".header-section-fade-in").fadeIn("slow");
    } else {
      $(".header-section-fade-in").fadeOut("slow");
    }
  })
</script>
</body>
</html>