<!DOCTYPE html>
<html>
<head>
    <!-- Standard Meta -->
    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

    <title>Defining Genome-wide TR Catalogs</title>
    <link rel="stylesheet" target="_blank" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css"
          integrity="sha512-8bHTC73gkZ7rZ7vpqUQThUDhqcNFyYi2xgDgPDHc+GXVGHXq+xPjynxIopALmOPqzo9JZj0k6OqqewdGO3EsrQ=="
          crossorigin="anonymous" referrerpolicy="no-referrer"/>
    <link rel="stylesheet" href="./github-css.css"/>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-G7WEXHL9KW"></script>
    <script>
        window.dataLayer = window.dataLayer || [];

        function gtag() {
            dataLayer.push(arguments);
        }

        gtag('js', new Date());

        gtag('config', 'G-G7WEXHL9KW');
    </script>
    <style>
        /* no wrap */
        td {
            text-align: right !important;
            white-space: nowrap;
        }
        th {
            text-align: center !important;
            white-space: nowrap;
        }

        .header-section {
            font-weight: 500;
            color: #555555;
            display: inline-block;
        }

        /* used by anchor-js */
        :target {
            scroll-margin-top: 70px;
        }

        /* tool comparison plot viewer */
        .ui.form {
            font-size: 16px;
        }

        .ui.form label {
            font-size: 16px;
        }

        body {
            overflow-x: scroll !important;
        }
    </style>
</head>
<body>

<a name="top"></a>

<div class="ui fixed menu" style="padding: 10px 0px">
    <div class="ui grid" style="width:100%; font-size:16px; color: rgba(0,0,0,.87)">
        <div class="ui three wide column"></div>
        <div class="ui three wide column">
            <div style="font-weight:800; padding-right:10px; display:inline-block"><a href="/">Project Notes</a></div>
            <div class="header-section"> - Ben Weisburd</div>
        </div>
        <div class="ui six wide column">
            <div class="header-section, header-section-fade-in" style="color: black; display: none; width: 100%">
                Defining a Genome-wide TR Catalog
            </div>
        </div>
        <div class="ui one wide column">
            <div class="header-section" style="float: right;">11/12/2023</div>
        </div>
        <div class="ui three wide column">
            <div class="header-section, header-section-fade-in" style="display: none;">[<a href="#table-of-contents">Jump
                to top</a>]
            </div>
        </div>
    </div>
</div>

<div class="ui grid">
    <div class="ui three wide column"></div>
    <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px">

        <h2>Defining a Genome-wide TR Catalog</h2>
        <p>Most existing tools for genotyping tandem repeats (TRs) from short-read or long-read sequencing data require the user
            to provide the exact start/end coordinates and motifs of all TR loci to genotype. Due to this, a key step in any
            genome-wide TR analysis is to define the loci to include (aka. the TR catalog).
        This choice directly affects sensitivity since any loci not in the catalog will be missing from the downstream analysis. Also, it
            determines compute costs since runtimes of most current tools are proportional to the number of loci being genotyped.
            There is no consensus in the literature on how to approach this choice, and recently published papers have used
            a wide range of catalogs.
        </p>
        <p>Catalogs can differ from each other in several important ways, including:</p>
        <ul>
            <li> total # of loci</li>
            <li> whether they include homopolymers</li>
            <li> their overall distribution of motif sizes</li>
            <li> whether they include interrupted repeats, and if yes, how often their locus definitions overlap each other</li>
        </ul>
        The table below compares publicly available catalogs. It counts repeat intervals (aka. loci) in each catalog after removing small numbers of erroneous records from several of the catalogs (such as homopolymers of 'N' rather than A/C/G/T). Also, some catalogs (such as the
        one with known disease-associated loci) define adjacent repeats - for example, FXN is defined as (A)*(GAA)*. I count any adjacent repeats independently in the stats below (which is why the catalog of known disease-assocaited TRs
        has a non-zero count of homopolymer loci).

        <br />
        as well as their other properties:
        <br />
        <table>
            <thead>
            <tr style="text-align: right;">
                <th>catalog name</th>
                <th>total loci</th>
                <th>chrX</th>
                <th>chrY</th>
                <th>motif sizes</ht>
                <th>locus sizes</th>
                <th># of repeats</th>
                <th>pure repeats</th>
                <th>trimmed</th>
                <th>overlapping loci</th>
            </tr>
            </thead>
            <tbody>
            <tr>
                <td>
                    <a href="https://github.com/broadinstitute/str-analysis/blob/main/str_analysis/variant_catalogs/variant_catalog_without_offtargets.GRCh38.json">
                    Known disease-associated loci
                    </a>
                </td>
                <td>72</td>
                <td>8 (11.3%)</td>
                <td>0 (0.0%)</td>
                <td>1-27bp</td>
                <td>12-150bp</td>
                <td>1-50x</td>
                <td>58.3%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://www.biorxiv.org/content/10.1101/2023.05.05.539588v1">TRF catalog: only pure repeats <br /> + homopolymers spanning ≥ 6bp in hg38<a/></td>
                <td>9,223,577</td>
                <td>489,574 (5.3%)</td>
                <td>0 (0.0%)</td>
                <td>1-50bp</td>
                <td>6-1597bp</td>
                <td>3-300x</td>
                <td>100.0%</td>
                <td>78.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td>GangSTR v17</td>
                <td>1,340,266</td>
                <td>65,719 (4.9%)</td>
                <td>8,979 (0.7%)</td>
                <td>1-20bp</td>
                <td>10-612bp</td>
                <td>3-248x</td>
                <td>100.0%</td>
                <td>100.0%</td>
                <td>16 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://www.biorxiv.org/content/10.1101/2023.05.05.539588v1">TRF catalog: pure &amp; interrupted<br /> repeats spanning ≥ 6bp in hg38<a/></td>
                <td>4,596,675</td>
                <td>244,906 (5.3%)</td>
                <td>0 (0.0%)</td>
                <td>2-50bp</td>
                <td>6-6800bp</td>
                <td>3-1091x</td>
                <td>85.1%</td>
                <td>100.0%</td>
                <td>84,822 (1.8%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/HipSTR-Tool/HipSTR-references">HipSTR</a></td>
                <td>1,634,957</td>
                <td>79,233 (4.8%)</td>
                <td>12,638 (0.8%)</td>
                <td>1-6bp</td>                
                <td>11-102545bp</td>
                <td>3-20509x</td>
                <td>46.6%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/PacificBiosciences/trgt/tree/main/repeats">TRGT</a></td>
                <td>171,146</td>
                <td>8,558 (5.0%)</td>
                <td>0 (0.0%)</td>
                <td>2-24bp</td>                
                <td>4-532bp</td>
                <td>2-133x</td>
                <td>95.5%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/Illumina/RepeatCatalogs">Illumina</a></td>
                <td>174,285</td>
                <td>8,561 (4.9%)</td>
                <td>0 (0.0%)</td>
                <td>2-24bp</td>
                <td>4-532bp</td>
                <td>2-133x</td>
                <td>95.5%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td>TR truth set <br />from 51 HPRC samples</td>
                <td>1,573,403</td>
                <td>70,885 (4.5%)</td>
                <td>4,806 (0.3%)</td>
                <td>1-50bp</td>
                <td>1-1591bp</td>
                <td>1-301x</td>
                <td>94.7%</td>
                <td>100.0%</td>
                <td>131,515 (8.4%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/DecodeGenetics/popSTR">popSTR</a></td>
                <td>5,401,399</td>
                <td>0 (0.0%)</td>
                <td>0 (0.0%)</td>
                <td>1-6bp</td>
                <td>11-137bp</td>
                <td>2-128x</td>
                <td>78.0%</td>
                <td>42.0%</td>
                <td>11,202 (0.2%)</td>
            </tr>                    
            <tr>
                <td><a href="https://zenodo.org/records/8387564">Adotto v1.2</a></td>
                <td>2,934,732</td>
                <td>109,740 (3.7%)</td>
                <td>16,585 (0.6%)</td>
                <td>2-495bp</td>
                <td>4-49816bp</td>
                <td>1-8836x</td>
                <td>41.6%</td>
                <td>32.0%</td>
                <td>26,198 (0.9%)</td>
            </tr>
            </tbody>
        </table>
        <br />
        Also, these catalogs vary a lot in their distributions of motif sizes: <br />
        <br />
        <img src="https://bw2.github.io/2023-11-12-defining-genome-wide-TR-catalogs/variant_catalog_stats.10_catalogs.motif_sizes.png?random=1" />
        
        <br />
        <h3>Combining catalogs:</h3>

        I think that currrently the best way to create a comprehensive genome-wide TR catalog is to merge:
        <ol>
            <li>several of the larger catalogs generated by running TRF on the reference with different parameters (such as the
                catalog of only pure repeats, and the catalog of repeats that allow mismatches but not indels). These catalogs
                should capture most of polymorphic loci, as well as loci that vary rarely (eg. ~1/100000 individuals), and
                so could be missed by considering only polymorphic loci identified based on a small number of samples.
            </li>
            <li>all available empirical catalogs - including the Illumina catalog and the TR truth set from the table above - since
                these were generated using orthogonal methods that don't require some minimum number of repeats in the
                reference genome. These catalogs will include polymorphic loci that have only 1 or 2 repeats in hg38,
                and so are not captured by TRF-based catalogs.
            </li>
        </ol>

        The <a href="https://github.com/broadinstitute/str-analysis">str-analysis</a> repo contains scripts for merging, annotating and filtering TR catalogs. The commands below shows how to use them to generate a combined TR catalog:

        <pre>
## download the publicly-available catalogs from the google buckets and repos associated with [weisburd 2023]
known_disease_associated_loci=https://raw.githubusercontent.com/broadinstitute/str-analysis/main/str_analysis/variant_catalogs/variant_catalog_without_offtargets.GRCh38.json
trf_pure_6bp_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_6bp.bed.gz
trf_pure_9bp_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_9bp.bed.gz
gangstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/hg38_ver17.adjusted.bed.gz
trgt_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/trgt_repeat_catalog.hg38.reformatted_to_motif_only.bed.gz
illumina_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/illumina_variant_catalog.sorted.bed.gz
truth_set_catalog=https://storage.cloud.google.com/str-truth-set-v2/results_with_homopolymers/results_from_combined/combined.51_samples.positive_loci.json
hipstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/hg38.hipstr_reference.adjusted.bed.gz
trf_6bp_catalog_with_mismatches=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_allowing_mismatches.sorted.trimmed.at_least_6bp.bed.gz
trf_9bp_catalog_with_mismatches=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_allowing_mismatches.sorted.trimmed.at_least_9bp.bed.gz

popstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/popstr_catalog_v2.bed.gz
adotto_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/adotto_tr_catalog_v1.2.bed.gz

ref_fasta=/local/path/hg38.fa
output_prefix=combined_catalog.trf_at_least_9bp

## annotate each catalog and discard loci with N in the motif
for path in ${known_disease_associated_loci} \
            ${trf_pure_9bp_catalog} \
            ${gangstr_catalog} \
            ${trf_9bp_catalog_with_mismatches} \
            ${hipstr_catalog} \
            ${trgt_catalog} \
            ${illumina_catalog} \
            ${truth_set_catalog} \
            ${popstr_catalog} \
            ${adotto_catalog};
do
    # download each catalog
    wget $path

    # annotate loci and discard edge cases like motif=N
    python3 -u -m str_analysis.annotate_and_filter_str_catalog  \
            -r ${ref_fasta} \
            --output-stats \
            --output-tsv \
            --verbose \
            --discard-loci-with-non-acgt-bases \
            ${path}
done

## merge all catalogs into a single file (excluding the popSTR and Adotto catalogs because they are unusual and/or suboptimal in some ways)
python3 -u -m str_analysis.combine_str_catalogs --verbose \
  --merge-adjacent-loci-with-same-motif \
  --output-prefix ${output_prefix} \
  --add-source-field \
  $(basename ${known_disease_associated_loci} | sed 's/.json/.annotated_and_filtered.json.gz/') \
  $(basename ${trf_pure_9bp_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${gangstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${trf_9bp_catalog_with_mismatches} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${hipstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${trgt_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${illumina_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
  $(basename ${truth_set_catalog} | sed 's/.json/.annotated_and_filtered.json.gz/')


## add adjacent loci to the catalog
python3 -u -m str_analysis.add_adjacent_loci_to_expansion_hunter_catalog \
  --ref-fasta ${ref_fasta} \
  --max-distance-between-adjacent-repeats 6 \
  --max-adjacent-repeats 2 \
  --source-of-adjacent-loci ~/code/str-truth-set/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_6bp.bed.gz \
  ${output_prefix}.json.gz

## annotate the combined catalog
python3 -u -m str_analysis.annotate_and_filter_str_catalog \
  -r ${ref_fasta} \
  --output-stats \
  --output-tsv \
  --verbose \
  --discard-loci-with-non-acgt-bases \
  ${output_prefix}.with_adjacent_loci.json.gz

## compute catalog stats
python3 -u -m str_analysis.compute_catalog_stats --verbose \
	$(basename ${known_disease_associated_loci} | sed 's/.json/.annotated_and_filtered.json.gz/') \
	$(basename ${trf_pure_6bp_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${gangstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${popstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${trf_6bp_catalog_with_mismatches} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${hipstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${trgt_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${illumina_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
	$(basename ${truth_set_catalog} | sed 's/.json/.annotated_and_filtered.json.gz/') \
	$(basename ${adotto_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/')

        </pre>

        Running these steps yields a combined catalog with the following properties:

        <pre>            
Stats for combined_catalog.trf_at_least_9bp.with_adjacent_loci.annotated_and_filtered.json.gz:
    4,918,783 total loci
        1,139,440 out of  4,918,783 ( 23.2%) loci have adjacent repeats
    6,196,806 total repeat intervals
        4,522,377 out of  6,196,806 ( 73.0%) repeat interval size is an integer multiple of the motif size (aka. trimmed)
        1,792,664 out of  6,196,806 ( 28.9%) repeat intervals are homopolymers
        1,534,528 out of  6,196,806 ( 24.8%) repeat intervals overlap each other by at least two motif lengths
Examples of overlapping repeats: chr19:22862456-22862510, chr16:87038648-87038702, chr6:131131459-131131467, chr3:75804442-75804495, chrX:53113326-53113335, chrX:141112874-141112888, chr14:36902984-36902999, chr12:6740055-6740065, chr16:7165935-7165979, chr15:91057135-91057191

Ranges:
   Motif size range: 1-50bp
   Locus size range: 1-102545bp
   Num repeats range: 1-20509x repeats

   Maximum locus size = 102545bp               @ chrY:11490351-11592896 (AATGG)
   Minimum fraction pure bases = 0.00      @ chrY:26670808-26670828 (AATGG)
   Minimum fraction pure repeats = 0.00    @ chrX:155370990-155371126 (TATATAACTATAAATAAGTATATATTTATATATT)
   Minimum overall mappability = 0.00       @ chrY:24489213-24489225 (GCCT)

          chrX:    322,966 out of  6,196,806 (  5.2%) repeat intervals
          chrY:     16,670 out of  6,196,806 (  0.3%) repeat intervals
          chrM:         15 out of  6,196,806 (  0.0%) repeat intervals
   alt contigs:          0 out of  6,196,806 (  0.0%) repeat intervals

Motif size distribution:
          1bp:  1,792,664 out of  6,196,806 ( 28.9%) repeat intervals
          2bp:  1,479,000 out of  6,196,806 ( 23.9%) repeat intervals
          3bp:  1,532,929 out of  6,196,806 ( 24.7%) repeat intervals
          4bp:    839,986 out of  6,196,806 ( 13.6%) repeat intervals
          5bp:    256,601 out of  6,196,806 (  4.1%) repeat intervals
          6bp:    126,362 out of  6,196,806 (  2.0%) repeat intervals
       7-24bp:    120,964 out of  6,196,806 (  2.0%) repeat intervals
        25+bp:     48,300 out of  6,196,806 (  0.8%) repeat intervals

Fraction pure bases distribution:
          0.0:      8,522 out of  6,196,806 (  0.1%) repeat intervals
          0.1:      8,595 out of  6,196,806 (  0.1%) repeat intervals
          0.2:     16,730 out of  6,196,806 (  0.3%) repeat intervals
          0.3:     16,591 out of  6,196,806 (  0.3%) repeat intervals
          0.4:     15,682 out of  6,196,806 (  0.3%) repeat intervals
          0.5:     36,774 out of  6,196,806 (  0.6%) repeat intervals
          0.6:     58,015 out of  6,196,806 (  0.9%) repeat intervals
          0.7:     82,843 out of  6,196,806 (  1.3%) repeat intervals
          0.8:    141,233 out of  6,196,806 (  2.3%) repeat intervals
          0.9:     71,488 out of  6,196,806 (  1.2%) repeat intervals
          1.0:  5,740,333 out of  6,196,806 ( 92.6%) repeat intervals

Mappability distribution:
          0.0:     30,591 out of  4,918,783 (  0.6%) loci
          0.1:     21,219 out of  4,918,783 (  0.4%) loci
          0.2:     27,035 out of  4,918,783 (  0.5%) loci
          0.3:     29,659 out of  4,918,783 (  0.6%) loci
          0.4:     38,958 out of  4,918,783 (  0.8%) loci
          0.5:     68,023 out of  4,918,783 (  1.4%) loci
          0.6:     51,683 out of  4,918,783 (  1.1%) loci
          0.7:     70,885 out of  4,918,783 (  1.4%) loci
          0.8:    110,040 out of  4,918,783 (  2.2%) loci
          0.9:    289,968 out of  4,918,783 (  5.9%) loci
          1.0:  4,180,722 out of  4,918,783 ( 85.0%) loci
</pre>

        <h3>IGV.js display of TR catalogs</h3>
        <br />
    <div class="ui three wide column"></div>
</div>

<br />
<p class="ui horizontal divider"></p>
<hr />
<br />

<div>[<a target="_blank" href="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html">Open IGV.js in new window</a>]</div>
<br/>
<iframe id="igv-iframe" src="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html" width="100%" height="1700px"
frameBorder="0" style="padding:0px; margin:0px 0px 10px 0px;"></iframe>

    <div class="ui grid">
        <div class="ui three wide column"></div>
        <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px;">

            <h2>Prior studies</h2>
            <p>[<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/">Willems 2014</a>], one of the first studies to perform a genome-wide TR analysis using Illumina short read sequencing data, put considerable effort into defining an appropriate catalog, as described in their <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/#sec-1title">Results section</a> and <a href="https://genome.cshlp.org/content/24/11/1894/suppl/DC1">Supplementary Materials</a> which are copy-pasted below.
                However, their approach was limited to the TR alleles observed within the hg19 reference genome. Also, they used simulations to calculate thresholds that distinguish biologically meaningful STRs from random or accidental repetative sequences, but this insilico approach is relatively blunt - just because a sequence could occur randomly with some probability doesn't mean it's not biologically important if it occurs invivo.</p>
            <p>Also, data has since become available from the Human Pangenome Research Consortium (HPRC) as well as from other studies such as T2T-YAO from [<a href="https://www.biorxiv.org/content/10.1101/2023.07.17.549286v1">He 2023</a>] and the <a href="https://github.com/Illumina/RepeatCatalogs">Illumina catalog by Qiu, Dolzhenko, et al.</a> that now allows us to directly / empirically identify polymorphic TR loci and study their properties.</p>
            <hr />

            <p>Relevant sections from the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/#sec-1title">Results</a> and <a href="https://genome.cshlp.org/content/24/11/1894/suppl/DC1">Supplementary Materials</a> of
                <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929">Willems T, Gymrek M, Highnam G; 1000 Genomes Project Consortium, Mittelman D, Erlich Y. The landscape of human STR variation. Genome Res. 2014</a>:<br />
                <br /></p>
            <p align="center"><kbd><img width="798" alt="image" src="https://github.com/bw2/bw2.github.io/assets/6240170/1f3cbc71-36b1-4c9e-9c57-7cf47763b3eb"></kbd></p>

            <p align="center"><kbd>
                <img width="798" alt="supplementary text" src="https://github.com/bw2/bw2.github.io/assets/6240170/a492bf1a-f56d-4923-a465-75226ceec7f0">
            </kbd></p>

            <br />
            <center><img style="margin-left:25px;" width="586" alt="image" src="https://user-images.githubusercontent.com/6240170/167290392-8974f18a-af18-40c1-9f61-fbb6713fc8d0.png"></center>

            <!-- https://giscus.app/ -->
            <div class="container giscus" style="width:100%; margin:50px 10px">
                <script src="https://giscus.app/client.js"
                        data-repo="bw2/bw2.github.io"
                        data-repo-id="MDEwOlJlcG9zaXRvcnkzNTU2OTQ0OA=="
                        data-category="Announcements"
                        data-category-id="DIC_kwDOAh6_KM4CO_Vz"
                        data-mapping="pathname"
                        data-reactions-enabled="1"
                        data-emit-metadata="0"
                        data-input-position="bottom"
                        data-theme="light"
                        data-lang="en"
                        crossorigin="anonymous"
                        async>
                </script>
            </div>
        </div>
        <div class="ui three wide column"></div>
    </div>

<!-- script section -->
<script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"
        integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js"
        integrity="sha512-dqw6X88iGgZlTsONxZK9ePmJEFrmHwpuMrsUChjAw1mRUhUITE5QU9pkcSox+ynfLhL15Sv2al5A0LVyDCmtUw=="
        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script>
    const headingAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left'
    });
    headingAnchors.add('h2, h3, h4');

    const paragraphAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left',
        icon: '¶'
    });
    paragraphAnchors.add('p');

    $('.ui.checkbox').checkbox();

    $(window).scroll(() => {
        if (window.scrollY > 75) {
            $(".header-section-fade-in").fadeIn("slow");
        } else {
            $(".header-section-fade-in").fadeOut("slow");
        }
    })
</script>
</body>
</html>
