<!DOCTYPE html>
<html>
<head>
    <!-- Standard Meta -->
    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

    <title>Defining Genome-wide TR Catalogs</title>
    <link rel="stylesheet" target="_blank" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css"
          integrity="sha512-8bHTC73gkZ7rZ7vpqUQThUDhqcNFyYi2xgDgPDHc+GXVGHXq+xPjynxIopALmOPqzo9JZj0k6OqqewdGO3EsrQ=="
          crossorigin="anonymous" referrerpolicy="no-referrer"/>
    <link rel="stylesheet" href="./github-css.css"/>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-G7WEXHL9KW"></script>
    <script>
        window.dataLayer = window.dataLayer || [];

        function gtag() {
            dataLayer.push(arguments);
        }

        gtag('js', new Date());

        gtag('config', 'G-G7WEXHL9KW');
    </script>
    <style>
        /* no wrap */
        td {
            text-align: right !important;
            white-space: nowrap;
        }
        th {
            text-align: center !important;
            white-space: nowrap;
        }

        .header-section {
            font-weight: 500;
            color: #555555;
            display: inline-block;
        }

        /* used by anchor-js */
        :target {
            scroll-margin-top: 70px;
        }

        /* tool comparison plot viewer */
        .ui.form {
            font-size: 16px;
        }

        .ui.form label {
            font-size: 16px;
        }

        body {
            overflow-x: scroll !important;
        }
    </style>
</head>
<body>

<a name="top"></a>

<div class="ui fixed menu" style="padding: 10px 0px">
    <div class="ui grid" style="width:100%; font-size:16px; color: rgba(0,0,0,.87)">
        <div class="ui three wide column"></div>
        <div class="ui three wide column">
            <div style="font-weight:800; padding-right:10px; display:inline-block"><a href="/">Project Notes</a></div>
            <div class="header-section"> - Ben Weisburd</div>
        </div>
        <div class="ui six wide column">
            <div class="header-section, header-section-fade-in" style="color: black; display: none; width: 100%">
                Defining a Genome-wide TR Catalog
            </div>
        </div>
        <div class="ui one wide column">
            <div class="header-section" style="float: right;">11/12/2023</div>
        </div>
        <div class="ui three wide column">
            <div class="header-section, header-section-fade-in" style="display: none;">[<a href="#table-of-contents">Jump
                to top</a>]
            </div>
        </div>
    </div>
</div>

<div class="ui grid">
    <div class="ui three wide column"></div>
    <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px">

        <h2>Defining a Genome-wide TR Catalog</h2>
        <p>Most existing tools for genotyping tandem repeats (TRs) from short-read or long-read sequencing data require the user
            to provide the exact start/end coordinates and motifs of all TR loci to genotype. Due to this, a key step in any
            genome-wide TR analysis is to define the loci to include (aka. the TR catalog).
        This choice directly affects sensitivity since any loci not in the catalog will be missing from the downstream analysis. Also, it
            determines compute costs since runtimes of most current tools are proportional to the number of loci being genotyped.
            There is no consensus in the literature on how to approach this choice, and recently published papers have used
            a wide range of catalogs.
        </p>
        <p>Catalogs can differ from each other in several important ways, including:</p>
        <ul>
            <li> total # of loci</li>
            <li> whether they include homopolymers</li>
            <li> their overall distribution of motif sizes</li>
            <li> whether they include interrupted repeats, and if yes, how often their locus definitions overlap each other</li>
        </ul>
        The table below compares publicly available catalogs. It counts repeat intervals (aka. loci) in each catalog after removing small numbers of erroneous records from several of the catalogs (such as homopolymers of 'N' rather than A/C/G/T). Also, some catalogs (such as the
        one with known disease-associated loci) define adjacent repeats - for example, FXN is defined as (A)*(GAA)*. I count any adjacent repeats independently in the stats below (which is why the catalog of known disease-assocaited TRs
        has a non-zero count of homopolymer loci).

        <br />
        as well as their other properties:
        <br />
        <table>
            <thead>
            <tr style="text-align: right;">
                <th>catalog name</th>
                <th>total loci</th>
                <th>chrX</th>
                <th>chrY</th>
                <th>motif sizes</ht>
                <th>locus sizes</th>
                <th># of repeats</th>
                <th>pure repeats</th>
                <th>trimmed</th>
                <th>overlapping loci</th>
            </tr>
            </thead>
            <tbody>
            <tr>
                <td>
                    <a href="https://github.com/broadinstitute/str-analysis/blob/main/str_analysis/variant_catalogs/variant_catalog_without_offtargets.GRCh38.json">
                    Known disease-associated loci
                    </a>
                </td>
                <td>72</td>
                <td>8 (11.3%)</td>
                <td>0 (0.0%)</td>
                <td>1-27bp</td>
                <td>12-150bp</td>
                <td>1-50x</td>
                <td>58.3%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://www.biorxiv.org/content/10.1101/2023.05.05.539588v1">TRF catalog: only pure repeats <br /> + homopolymers spanning ≥ 6bp in hg38<a/></td>
                <td>9,223,577</td>
                <td>489,574 (5.3%)</td>
                <td>0 (0.0%)</td>
                <td>1-50bp</td>
                <td>6-1597bp</td>
                <td>3-300x</td>
                <td>100.0%</td>
                <td>78.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td>GangSTR v17</td>
                <td>1,340,266</td>
                <td>65,719 (4.9%)</td>
                <td>8,979 (0.7%)</td>
                <td>1-20bp</td>
                <td>10-612bp</td>
                <td>3-248x</td>
                <td>100.0%</td>
                <td>100.0%</td>
                <td>16 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://www.biorxiv.org/content/10.1101/2023.05.05.539588v1">TRF catalog: pure &amp; interrupted<br /> repeats spanning ≥ 6bp in hg38<a/></td>
                <td>4,596,675</td>
                <td>244,906 (5.3%)</td>
                <td>0 (0.0%)</td>
                <td>2-50bp</td>
                <td>6-6800bp</td>
                <td>3-1091x</td>
                <td>85.1%</td>
                <td>100.0%</td>
                <td>84,822 (1.8%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/HipSTR-Tool/HipSTR-references">HipSTR</a></td>
                <td>1,634,957</td>
                <td>79,233 (4.8%)</td>
                <td>12,638 (0.8%)</td>
                <td>1-6bp</td>                
                <td>11-102545bp</td>
                <td>3-20509x</td>
                <td>46.6%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/PacificBiosciences/trgt/tree/main/repeats">TRGT</a></td>
                <td>171,146</td>
                <td>8,558 (5.0%)</td>
                <td>0 (0.0%)</td>
                <td>2-24bp</td>                
                <td>4-532bp</td>
                <td>2-133x</td>
                <td>95.5%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/Illumina/RepeatCatalogs">Illumina</a></td>
                <td>174,285</td>
                <td>8,561 (4.9%)</td>
                <td>0 (0.0%)</td>
                <td>2-24bp</td>
                <td>4-532bp</td>
                <td>2-133x</td>
                <td>95.5%</td>
                <td>100.0%</td>
                <td>0 (0.0%)</td>
            </tr>
            <tr>
                <td>TR truth set <br />from 51 HPRC samples</td>
                <td>1,573,403</td>
                <td>70,885 (4.5%)</td>
                <td>4,806 (0.3%)</td>
                <td>1-50bp</td>
                <td>1-1591bp</td>
                <td>1-301x</td>
                <td>94.7%</td>
                <td>100.0%</td>
                <td>131,515 (8.4%)</td>
            </tr>
            <tr>
                <td><a href="https://github.com/DecodeGenetics/popSTR">popSTR</a></td>
                <td>5,401,399</td>
                <td>0 (0.0%)</td>
                <td>0 (0.0%)</td>
                <td>1-6bp</td>
                <td>11-137bp</td>
                <td>2-128x</td>
                <td>78.0%</td>
                <td>42.0%</td>
                <td>11,202 (0.2%)</td>
            </tr>                    
            <tr>
                <td><a href="https://zenodo.org/records/8387564">Adotto v1.2</a></td>
                <td>2,934,732</td>
                <td>109,740 (3.7%)</td>
                <td>16,585 (0.6%)</td>
                <td>2-495bp</td>
                <td>4-49816bp</td>
                <td>1-8836x</td>
                <td>41.6%</td>
                <td>32.0%</td>
                <td>26,198 (0.9%)</td>
            </tr>
            </tbody>
        </table>
        <br />
        Also, the catalogs vary widely in their distribution of motif sizes:
        
        
        <br />
        <h3>Combining catalogs:</h3>

        I think the best way to create a comprehensive genome-wide TR catalog is to merge:
        <ol>
            <li>catalog(s) generated by running TRF on the reference</li>
            <li>all available empirical catalogs - such as the Illumina catalog and TR truth set in the above table - since
                these are generated using orthogonal approaches that don't require some minimum number of repeats to be
                present in the reference genome.
            </li>
        </ol>
        By merging these two types of catalogs, we can capture:
        <ul>
            <li>polymorphic TR loci that have only one or two repeats in
            the hg38 reference and so can only be captured by these empirical catalogs and not by TRF</li>
            <li>loci where variation is very rare and so would likely be missed by empirical approaches
                that only consider polymorphic loci</li>
        </ul>

        The str-analysis repo contains scripts for merging, annotating and filtering TR catalogs. The shell script
        below demonstrates how to use them to generate a comprehensive TR catalog:

        <pre>
## download various publicly-available catalogs. Some of these have been slightly reformatted from the original.

known_disease_associated_loci=https://raw.githubusercontent.com/broadinstitute/str-analysis/main/str_analysis/variant_catalogs/variant_catalog_without_offtargets.GRCh38.json
trf_pure_6bp_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_6bp.bed.gz
trf_pure_9bp_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_9bp.bed.gz
gangstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/hg38_ver17.adjusted.bed.gz
trgt_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/trgt_repeat_catalog.hg38.reformatted_to_motif_only.bed.gz
illumina_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/illumina_variant_catalog.sorted.bed.gz
truth_set_catalog=https://storage.cloud.google.com/str-truth-set-v2/results_with_homopolymers/results_from_combined/combined.51_samples.positive_loci.json
hipstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/hg38.hipstr_reference.adjusted.bed.gz
trf_6bp_catalog_with_mismatches=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_allowing_mismatches.sorted.trimmed.at_least_6bp.bed.gz
trf_9bp_catalog_with_mismatches=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/repeat_specs_GRCh38_allowing_mismatches.sorted.trimmed.at_least_9bp.bed.gz

popstr_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/popstr_catalog_v2.bed.gz
adotto_catalog=https://storage.cloud.google.com/str-truth-set/hg38/ref/other/adotto_tr_catalog_v1.2.bed.gz

ref_fasta=/local/path/hg38.fa
output_prefix=combined_catalog.trf_at_least_9bp

## annotate each catalog and discard loci with N in the motif
for path in ${known_disease_associated_loci} \
            ${trf_pure_9bp_catalog} \
            ${gangstr_catalog} \
            ${trf_9bp_catalog_with_mismatches} \
            ${hipstr_catalog} \
            ${trgt_catalog} \
            ${illumina_catalog} \
            ${truth_set_catalog} \
            ${popstr_catalog} \
            ${adotto_catalog};
do
    # download each catalog
    wget $path

    # annotate loci and discard edge cases like motif=N
    python3 -u -m str_analysis.annotate_and_filter_str_catalog  \
            -r ${ref_fasta} \
            --output-stats \
            --output-tsv \
            --verbose \
            --discard-loci-with-non-acgt-bases \
            ${path}
done

## merge all catalogs into a single file (excluding the popSTR and Adotto catalogs because they are unusual and/or suboptimal in some ways)
python3 -u -m str_analysis.combine_str_catalogs --verbose \
            --merge-adjacent-loci-with-same-motif \
            --output-prefix ${output_prefix} \
            --add-source-field \
            $(basename ${known_disease_associated_loci} | sed 's/.json/.annotated_and_filtered.json.gz/') \
            $(basename ${trf_pure_9bp_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${gangstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${trf_9bp_catalog_with_mismatches} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${hipstr_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${trgt_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${illumina_catalog} | sed 's/.bed.gz/.annotated_and_filtered.json.gz/') \
            $(basename ${truth_set_catalog} | sed 's/.json/.annotated_and_filtered.json.gz/')

## annotate the combined catalog
python3 -u -m str_analysis.annotate_and_filter_str_catalog \
            -r ${ref_fasta} \
            --output-stats \
            --output-tsv \
            --verbose \
            --discard-loci-with-non-acgt-bases \
            ${output_prefix}.json.gz

## add adjacent loci to the catalog
python3 -u -m str_analysis.add_adjacent_loci_to_expansion_hunter_catalog \
  --ref-fasta ${ref_fasta} \
  --max-distance-between-adjacent-repeats 6 \
  --max-adjacent-repeats 2 \
  --source-of-adjacent-loci ~/code/str-truth-set/ref/other/repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_6bp.bed.gz \
  ${output_prefix}.annotated_and_filtered.json.gz

        </pre>

        Running these steps yields a combined catalog with the following properties:

        <pre>            
            Stats for combined_catalog.trf_at_least_9bp.annotated_and_filtered.json.gz:            
            4,944,109 total repeat intervals
            1,425,773 out of  4,944,109 ( 28.8%) repeat intervals are homopolymers
              160,509 out of  4,944,109 (  3.2%) repeat intervals overlap each other by at least two motif lengths
        Examples of overlapping repeats: chrX:8832362-8832422, chr5:140543682-140543702, chr6:469004-469232, chr5:26217645-26217668, chr1:245117424-245117439, chr17:42801508-42801544, chr12:111942768-111942782, chr2:198055294-198055330, chr5:43826764-43826783, chr10:83131202-83131230

        Ranges:
           Motif size range: 1-50bp
           Locus size range: 1-102545bp
           Num repeats range: 1-20509x repeats

           Maximum locus size = 102545bp               @ chrY:11490351-11592896 (AATGG)
           Minimum fraction pure bases = 0.00      @ chrY:26670808-26670828 (AATGG)
           Minimum fraction pure repeats = 0.00    @ chr11:135041141-135041273 (GTGTGTCCTGAGTCGGCATCTCCCCGTCACCAG)
           Minimum overall mappability = 0.00       @ chrY:24489213-24489225 (GCCT)

                  chrX:    252,353 out of  4,944,109 (  5.1%) repeat intervals
                  chrY:     16,670 out of  4,944,109 (  0.3%) repeat intervals
                  chrM:         15 out of  4,944,109 (  0.0%) repeat intervals
           alt contigs:          0 out of  4,944,109 (  0.0%) repeat intervals

        Motif size distribution:
                  1bp:  1,425,773 out of  4,944,109 ( 28.8%) repeat intervals
                  2bp:  1,033,483 out of  4,944,109 ( 20.9%) repeat intervals
                  3bp:  1,423,329 out of  4,944,109 ( 28.8%) repeat intervals
                  4bp:    651,610 out of  4,944,109 ( 13.2%) repeat intervals
                  5bp:    198,852 out of  4,944,109 (  4.0%) repeat intervals
                  6bp:    102,552 out of  4,944,109 (  2.1%) repeat intervals
               7-24bp:     83,580 out of  4,944,109 (  1.7%) repeat intervals
                25+bp:     24,930 out of  4,944,109 (  0.5%) repeat intervals

        Fraction pure bases distribution:
                  0.0:     10,598 out of  4,944,109 (  0.2%) repeat intervals
                  0.1:      6,701 out of  4,944,109 (  0.1%) repeat intervals
                  0.2:     14,438 out of  4,944,109 (  0.3%) repeat intervals
                  0.3:     16,835 out of  4,944,109 (  0.3%) repeat intervals
                  0.4:     16,413 out of  4,944,109 (  0.3%) repeat intervals
                  0.5:     38,659 out of  4,944,109 (  0.8%) repeat intervals
                  0.6:     58,094 out of  4,944,109 (  1.2%) repeat intervals
                  0.7:     82,071 out of  4,944,109 (  1.7%) repeat intervals
                  0.8:    140,936 out of  4,944,109 (  2.9%) repeat intervals
                  0.9:     71,735 out of  4,944,109 (  1.5%) repeat intervals
                  1.0:  4,487,629 out of  4,944,109 ( 90.8%) repeat intervals

        Wrote 4,918,796 total records to combined_catalog.trf_at_least_9bp.annotated_and_filtered.with_adjacent_loci.json
        Added adjacent loci to 1,164,753 out of 4,944,101 records (23.6%)

            Source of loci in output catalog:
   4,431,801 out of 4,992,309 (88.8%) repeat_specs_GRCh38_without_mismatches.including_homopolymers.sorted.at_least_9bp.annotated_and_filtered.json.gz
     287,561 out of 4,992,309 ( 5.8%) repeat_specs_GRCh38_allowing_mismatches.sorted.trimmed.at_least_9bp.annotated_and_filtered.json.gz
     158,585 out of 4,992,309 ( 3.2%) combined.51_samples.positive_loci.annotated_and_filtered.json.gz
      95,196 out of 4,992,309 ( 1.9%) hg38.hipstr_reference.adjusted.annotated_and_filtered.json.gz
       9,863 out of 4,992,309 ( 0.2%) trgt_repeat_catalog.hg38.reformatted_to_motif_only.annotated_and_filtered.json.gz
       9,064 out of 4,992,309 ( 0.2%) hg38_ver17.adjusted.annotated_and_filtered.json.gz
         175 out of 4,992,309 ( 0.0%) illumina_variant_catalog.sorted.annotated_and_filtered.json.gz
          64 out of 4,992,309 ( 0.0%) variant_catalog_without_offtargets.GRCh38.annotated_and_filtered.json.gz            
        </pre>

        <h3>IGV.js display of TR catalogs</h3>
        <br />
    <div class="ui three wide column"></div>
</div>

<br />
<p class="ui horizontal divider"></p>
<hr />
<br />

<div style="float:right">[<a target="_blank" href="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html">Open IGV.js in new window</a>]</div>
<br/>
<iframe id="igv-iframe" src="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html" width="100%" height="1700px"
frameBorder="0" style="padding:0px; margin:0px 0px 10px 0px;"></iframe>

    <div class="ui grid">
        <div class="ui three wide column"></div>
        <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px;">

            <h2>Prior studies</h2>
            <p>[<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/">Willems 2014</a>], one of the first studies to perform a genome-wide TR analysis using Illumina short read sequencing data, put considerable effort into defining an appropriate catalog, as described in their <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/#sec-1title">Results section</a> and <a href="https://genome.cshlp.org/content/24/11/1894/suppl/DC1">Supplementary Materials</a> which are copy-pasted below. One limitation of their analysis was that it was based only on the hg19 reference. </p>
            <p>Data from the Human Pangenome Research Consortium (HPRC) as well as other studies such as T2T-YAO from [<a href="https://www.biorxiv.org/content/10.1101/2023.07.17.549286v1">He 2023</a>] and the <a href="https://github.com/Illumina/RepeatCatalogs">Illumina catalog by Qiu, Dolzhenko, et al.</a> which contains 174k STR loci that are polymorphic within 1kGP.</p>
            <p>The many genome-wide TR analyses published since then have often used different catalogs. </p>
            <hr />

            <p>Relevant sections from the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929/#sec-1title">Results</a> and <a href="https://genome.cshlp.org/content/24/11/1894/suppl/DC1">Supplementary Materials</a> of
                <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4216929">Willems T, Gymrek M, Highnam G; 1000 Genomes Project Consortium, Mittelman D, Erlich Y. The landscape of human STR variation. Genome Res. 2014</a>:<br />
                <br /></p>
            <p align="center"><kbd><img width="798" alt="image" src="https://github.com/bw2/bw2.github.io/assets/6240170/1f3cbc71-36b1-4c9e-9c57-7cf47763b3eb"></kbd></p>

            <p align="center"><kbd>
                <img width="798" alt="supplementary text" src="https://github.com/bw2/bw2.github.io/assets/6240170/a492bf1a-f56d-4923-a465-75226ceec7f0">
            </kbd></p>

            <br />
            <center><img style="margin-left:25px;" width="586" alt="image" src="https://user-images.githubusercontent.com/6240170/167290392-8974f18a-af18-40c1-9f61-fbb6713fc8d0.png"></center>

            <!-- https://giscus.app/ -->
            <div class="container giscus" style="width:100%; margin:50px 10px">
                <script src="https://giscus.app/client.js"
                        data-repo="bw2/bw2.github.io"
                        data-repo-id="MDEwOlJlcG9zaXRvcnkzNTU2OTQ0OA=="
                        data-category="Announcements"
                        data-category-id="DIC_kwDOAh6_KM4CO_Vz"
                        data-mapping="pathname"
                        data-reactions-enabled="1"
                        data-emit-metadata="0"
                        data-input-position="bottom"
                        data-theme="light"
                        data-lang="en"
                        crossorigin="anonymous"
                        async>
                </script>
            </div>
        </div>
        <div class="ui three wide column"></div>
    </div>

<!-- script section -->
<script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"
        integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js"
        integrity="sha512-dqw6X88iGgZlTsONxZK9ePmJEFrmHwpuMrsUChjAw1mRUhUITE5QU9pkcSox+ynfLhL15Sv2al5A0LVyDCmtUw=="
        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script>
    const headingAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left'
    });
    headingAnchors.add('h2, h3, h4');

    const paragraphAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left',
        icon: '¶'
    });
    paragraphAnchors.add('p');

    $('.ui.checkbox').checkbox();

    $(window).scroll(() => {
        if (window.scrollY > 75) {
            $(".header-section-fade-in").fadeIn("slow");
        } else {
            $(".header-section-fade-in").fadeOut("slow");
        }
    })
</script>
</body>
</html>
