<!DOCTYPE html>
<html>
<head>
    <!-- Standard Meta -->
    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

    <title>Defining a Genome-wide TR Catalog</title>
    <link rel="stylesheet" target="_blank" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css"
          integrity="sha512-8bHTC73gkZ7rZ7vpqUQThUDhqcNFyYi2xgDgPDHc+GXVGHXq+xPjynxIopALmOPqzo9JZj0k6OqqewdGO3EsrQ=="
          crossorigin="anonymous" referrerpolicy="no-referrer"/>
    <link rel="stylesheet" href="./github-css.css"/>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-G7WEXHL9KW"></script>
    <script>
        window.dataLayer = window.dataLayer || [];

        function gtag() {
            dataLayer.push(arguments);
        }

        gtag('js', new Date());

        gtag('config', 'G-G7WEXHL9KW');
    </script>
    <style>
        /* no wrap */
        td {
            text-align: right !important;
            white-space: nowrap;
        }
        th {
            text-align: center !important;
            white-space: nowrap;
        }

        .header-section {
            font-weight: 500;
            color: #555555;
            display: inline-block;
        }

        /* used by anchor-js */
        :target {
            scroll-margin-top: 70px;
        }

        /* tool comparison plot viewer */
        .ui.form {
            font-size: 16px;
        }

        .ui.form label {
            font-size: 16px;
        }

        body {
            overflow-x: scroll !important;
        }
    </style>
</head>
<body>

<a name="top"></a>

<div class="ui fixed menu" style="padding: 10px 0px">
    <div class="ui grid" style="width:100%; font-size:16px; color: rgba(0,0,0,.87)">
        <div class="ui three wide column"></div>
        <div class="ui three wide column">
            <div style="font-weight:800; padding-right:10px; display:inline-block"><a href="/">Project Notes</a></div>
            <div class="header-section"> - Ben Weisburd</div>
        </div>
        <div class="ui six wide column">
            <div class="header-section, header-section-fade-in" style="color: black; display: none; width: 100%">
                Defining a Genome-wide TR Catalog
            </div>
        </div>
        <div class="ui one wide column">
            <div class="header-section" style="float: right;">11/12/2023</div>
        </div>
        <div class="ui three wide column">
            <div class="header-section, header-section-fade-in" style="display: none;">[<a href="#table-of-contents">Jump
                to top</a>]
            </div>
        </div>
    </div>
</div>

<div class="ui grid">
    <div class="ui three wide column"></div>
    <div class="ui ten wide column" style="font-size:16px; color: rgba(0,0,0,.87); padding-top:35px">

        <h2>Defining a Genome-wide TR Catalog</h2>
        <p>Most existing methods for genotpying tandem repeats (TRs) in short-read or long-read sequencing data require the user
            to specify the exact start and end coordinates and motifs of all TR loci to genotype. Therefore, a key step in any
            genome-wide analysis involves selecting the set of TRs (aka. catalog) to include.
        This choice directly affects sensitivity since any loci not in the catalog will be missing from downstream analysis, but also
            compute costs since runtimes of most current tools are proportional to the size of the catalog being genotyped.
            There is no consensus in the literature on how to approach this choice, and recently published papers have contiued to use
            a wide range of catalogs.
        </p>
        <p>Catalogs can differ from each other on several important properties, including:</p>
        <ul>
            <li> total # of loci</li>
            <li> includes homopolymers?</li>
            <li> distribution of motif sizes</li>
            <li> includes only pure, or also impure repeats?  and if it includes impure, how often do locus definitions overlap?</li>
            <li> do loci overlap each other?</li>
        </ul>
        The table and IGV.js visualization below compare publicly available catalogs:
    <div class="ui three wide column"></div>
</div>
<table>
    <thead>
    <tr style="text-align: right;">
        <th>catalog name</th>
        <th>total loci</th>
        <th>chrX</th>
        <th>chrY</th>
        <th>motif sizes</th>
        <th>locus sizes</th>
        <th># of repeats</th>
        <th>homopolymers</th>
        <th>2bp motifs</th>
        <th>3bp motifs</th>
        <th>4bp motifs</th>
        <th>5bp motifs</th>
        <th>6bp motifs</th>
        <th>7+bp motifs</th>
        <th>pure repeats</th>
        <th>trimmed</th>
        <th>overlapping loci</th>
        <th>source</th>
        <th>notes</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td>Known disease-associated loci</td>
        <td>71</td>
        <td>8 (11.3%)</td>
        <td>0 (0.0%)</td>
        <td>1-27bp</td>
        <td>12-150bp</td>
        <td>1-50x</td>
        <td>1.4%</td>
        <td>1.4%</td>
        <td>69.0%</td>
        <td>2.8%</td>
        <td>14.1%</td>
        <td>4.2%</td>
        <td>7.0%</td>
        <td>59.2%</td>
        <td>100.0%</td>
        <td>0 (0.0%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>New TRF: pure loci <br />including homopolymers</td>
        <td>9,237,117</td>
        <td>489,574 (5.3%)</td>
        <td>0 (0.0%)</td>
        <td>1-50bp</td>
        <td>6-1597bp</td>
        <td>3-300x</td>
        <td>47.8%</td>
        <td>28.3%</td>
        <td>15.1%</td>
        <td>6.0%</td>
        <td>1.8%</td>
        <td>0.6%</td>
        <td>0.4%</td>
        <td>99.9%</td>
        <td>78.0%</td>
        <td>0 (0.0%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>GangSTR ver17</td>
        <td>1,340,266</td>
        <td>65,719 (4.9%)</td>
        <td>8,979 (0.7%)</td>
        <td>1-20bp</td>
        <td>10-612bp</td>
        <td>3-248x</td>
        <td>36.5%</td>
        <td>13.4%</td>
        <td>11.2%</td>
        <td>28.2%</td>
        <td>7.9%</td>
        <td>2.1%</td>
        <td>0.8%</td>
        <td>100.0%</td>
        <td>100.0%</td>
        <td>16 (0.0%)</td>
    </tr>
    <tr>
        <td>popSTR</td>
        <td>5,401,401</td>
        <td>0 (0.0%)</td>
        <td>0 (0.0%)</td>
        <td>1-6bp</td>
        <td>11-137bp</td>
        <td>2-128x</td>
        <td>15.2%</td>
        <td>7.3%</td>
        <td>6.0%</td>
        <td>15.3%</td>
        <td>26.0%</td>
        <td>30.3%</td>
        <td>0.0%</td>
        <td>78.0%</td>
        <td>42.0%</td>
        <td>11,202 (0.2%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>New TRF: pure &amp; interrupted loci<br />excluding homopolymers</td>
        <td>4,604,776</td>
        <td>244,906 (5.3%)</td>
        <td>0 (0.0%)</td>
        <td>2-50bp</td>
        <td>6-6800bp</td>
        <td>3-1091x</td>
        <td>0.0%</td>
        <td>50.3%</td>
        <td>30.3%</td>
        <td>12.4%</td>
        <td>3.6%</td>
        <td>1.4%</td>
        <td>1.9%</td>
        <td>84.9%</td>
        <td>100.0%</td>
        <td>84,857 (1.8%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>HipSTR</td>
        <td>1,638,945</td>
        <td>79,235 (4.8%)</td>
        <td>16,003 (1.0%)</td>
        <td>1-6bp</td>
        <td>9-132210bp</td>
        <td>3-26442x</td>
        <td>51.3%</td>
        <td>18.3%</td>
        <td>4.9%</td>
        <td>14.7%</td>
        <td>6.2%</td>
        <td>4.5%</td>
        <td>0.0%</td>
        <td>46.5%</td>
        <td>100.0%</td>
        <td>18 (0.0%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>TRGT</td>
        <td>171,146</td>
        <td>8,558 (5.0%)</td>
        <td>0 (0.0%)</td>
        <td>2-24bp</td>
        <td>4-532bp</td>
        <td>2-133x</td>
        <td>0.0%</td>
        <td>56.7%</td>
        <td>8.3%</td>
        <td>24.8%</td>
        <td>7.2%</td>
        <td>2.3%</td>
        <td>0.7%</td>
        <td>95.5%</td>
        <td>100.0%</td>
        <td>0 (0.0%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>Illumina</td>
        <td>174,286</td>
        <td>8,561 (4.9%)</td>
        <td>0 (0.0%)</td>
        <td>2-24bp</td>
        <td>4-532bp</td>
        <td>2-133x</td>
        <td>0.0%</td>
        <td>56.7%</td>
        <td>8.3%</td>
        <td>24.8%</td>
        <td>7.2%</td>
        <td>2.3%</td>
        <td>0.7%</td>
        <td>95.5%</td>
        <td>100.0%</td>
        <td>0 (0.0%)</td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <td>TR truth set <br />from 51 HPRC samples</td>
        <td>1,573,403</td>
        <td>70,885 (4.5%)</td>
        <td>4,806 (0.3%)</td>
        <td>1-50bp</td>
        <td>1-1591bp</td>
        <td>1-301x</td>
        <td>65.0%</td>
        <td>15.4%</td>
        <td>4.1%</td>
        <td>10.2%</td>
        <td>2.2%</td>
        <td>1.0%</td>
        <td>2.1%</td>
        <td>94.7%</td>
        <td>100.0%</td>
        <td>131,515 (8.4%)</td>
        <td></td>
        <td></td>
    </tr>
    </tbody>
</table>

    <br />
    <br />
    <p class="ui horizontal divider"></p>
<hr />
<br />

<div style="float:right">[<a target="_blank" href="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html">Open IGV.js in new window</a>]</div>

<br/>
<iframe src="https://broadinstitute.github.io/str-truth-set/html/tgg_viewer_catalogs_only.html" width="100%" height="1500px"
frameBorder="0" style="padding:0px; margin:0px 0px 10px 0px;"></iframe>

<!-- script section -->
<script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script>
<script src="https://code.jquery.com/jquery-3.1.1.min.js"
        integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js"
        integrity="sha512-dqw6X88iGgZlTsONxZK9ePmJEFrmHwpuMrsUChjAw1mRUhUITE5QU9pkcSox+ynfLhL15Sv2al5A0LVyDCmtUw=="
        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
<script>
    const headingAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left'
    });
    headingAnchors.add('h2, h3, h4');

    const paragraphAnchors = new AnchorJS({
        visible: 'hover',
        placement: 'left',
        icon: 'Â¶'
    });
    paragraphAnchors.add('p');

    $('.ui.checkbox').checkbox();

    $(window).scroll(() => {
        if (window.scrollY > 75) {
            $(".header-section-fade-in").fadeIn("slow");
        } else {
            $(".header-section-fade-in").fadeOut("slow");
        }
    })
</script>
</body>
</html>
